src/Item/Item.router.spec.ts

```diff
@@ -315,37 +316,72 @@ describe('Item router', () => {
       ;(ItemCuration.find as jest.Mock).mockResolvedValueOnce([dbItemCuration])
       mockItemConsolidation([dbItem], [wearable])
       ;(peerAPI.fetchWearables as jest.Mock).mockResolvedValueOnce([tpWearable])
-      url = `/${wallet.address}/items`
     })
 
-    it('should return all the items of an address that are published with URN and the ones that are not without it', () => {
-      return server
-        .get(buildURL(url))
-        .set(createAuthHeaders('get', url))
-        .expect(200)
-        .then((response: any) => {
-          expect(response.body).toEqual({
-            data: [
-              {
-                ...resultingItem,
-                beneficiary: itemFragment.beneficiary,
-                collection_id: dbItem.collection_id,
-                blockchain_item_id: dbItem.blockchain_item_id,
-                urn: itemFragmentMock.urn,
-              },
-              resultItemNotPublished,
-              resultingTPItem,
-            ],
-            ok: true,
+    describe('and there pagination params are not passed', () => {
+      beforeEach(() => {
+        url = `/${wallet.address}/items`
+      })
+      it('should return all the items of an address that are published with URN and the ones that are not without it', () => {
+        return server
+          .get(buildURL(url))
+          .set(createAuthHeaders('get', url))
+          .expect(200)
+          .then((response: any) => {
+            expect(response.body).toEqual({
+              data: [
+                {
+                  ...resultingItem,
+                  beneficiary: itemFragment.beneficiary,
+                  collection_id: dbItem.collection_id,
+                  blockchain_item_id: dbItem.blockchain_item_id,
+                  urn: itemFragmentMock.urn,
+                },
+                resultItemNotPublished,
+                resultingTPItem,
+              ],
+              ok: true,
+            })
+            expect(Item.findAllItemsByAddress).toHaveBeenCalledWith(
+              [thirdPartyFragmentMock.id],
+              wallet.address,
+              undefined,
+              undefined
+            )
           })
-        })
+      })
+    })
+
+    describe('and pagination params are passed', () => {
+      let baseUrl: string
+      let page: number
+      let limit: number
+      beforeEach(() => {
+        ;(page = 1), (limit = 1)
+        baseUrl = `/${wallet.address}/items`
+        url = `${baseUrl}?limit=${limit}&page=${page}`
+      })
+      it('should call the find method with the pagination params', () => {
+        return server
+          .get(buildURL(url))
+          .set(createAuthHeaders('get', baseUrl))
+          .expect(200)
+          .then(() => {
+            expect(Item.findAllItemsByAddress).toHaveBeenCalledWith(
+              [thirdPartyFragmentMock.id],
+              wallet.address,
+              limit,
+              page - 1 // it's the offset
+            )
+          })
```

Comment:
This test is missing the body check, that is, the response, that includes all of the pagination data is not being checked here.